### PR TITLE
Added switch UI to enable anonymous state of topic

### LINKED ISSUE #8: Implement Anonymous Posting Option switch
User story: Anonymous post function 

Create a feature that allows users to toggle between posting anonymously or with their identity.
Add a button or checkbox in the post creation interface to enable this option.

Estimated effort:
~ 4 hrs

Dependencies: No dependency for the UI, but the connection to the Model will require the actual functionality to be working properly, menaing that issue #10 should be finished before we make the connection.

Acceptance criteria: Manual test on local server

For sprint 1, the function would be implemented front-end based. Further features like different names and database might be postponed to sprint 2 or stretched

### PR DESCRIPTION
**Pull request by EmmaShi4517**
Changes:

- Added a ui switch in the composer module (patch package)
- Added fields to the topic object (schema, model)
- Tried to have the switch linked to the anonymous state

Tests:

- Local lint and tests passed - added one simple test on topic creation with anonymous field
- Manual test: create topic w/wo anonymous switch turned on on local server and examined api output to find the anonymous field of topic

Resolves #8

### CODE REVIEW

File: patches/nodebb-plugin-composer-default+9.2.4.patch [hunk 1]
```diff
@@ -74,67 +75,144 @@ index e9a8ca6..ab81fb5 100644
...
  		};
  
  		({ pushData } = await hooks.fire('filter:composer.topic.push', {
-@@ -431,6 +432,7 @@ define('composer', [
+@@ -343,7 +345,9 @@ define('composer', [
+ 		submitBtn.on('click', function (e) {
+ 			e.preventDefault();
+ 			e.stopPropagation();	// Other click events bring composer back to active state which is undesired on submit
+-
++      if ($('.isAnon').is(':checked')) {
```
Comment (kaitlyncliu) [inline patches/nodebb-plugin-composer-default+9.2.4.patch]: This seems like logs for debugging, maybe remove for production

File: test/topics.js [hunk 2]
```diff
@@ -82,6 +83,22 @@ describe('Topic\'s', () => {
             });
         });
 
+        it('should create anonymous topic', (done) => {
```
Comment (kaitlyncliu) [inline test/topics.js:86]: Can you also add a test for a topic that isn't anonymous

**Review by kaitlyncliu — CHANGES_REQUESTED** [review]
LGTM other than the two comments I made!

File: patches/nodebb-plugin-composer-default+9.2.4.patch (hunk 1 above)
Comment (EmmaShi4517) [reply patches/nodebb-plugin-composer-default+9.2.4.patch]: Removed it from the package.

**Review by kaitlyncliu — APPROVED** [review]
LGTM thanks for making the changes!

**Comment by kellyycha** [comment]
Not using this feature anymore

**Pull request closed without merging**